turbopack/crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -553,6 +488,112 @@ impl SourceMap {
+                        if map.get_source_count() == 1 {
+                            let source = map.sources().next().unwrap();
+                            *guessed_original_file = Some(source.to_string());
+                        }
+                    }
+                }
+
+                if need_source_content && content.is_none() {
+                    if let Some(map) = map.map.as_regular_source_map() {
+                        content = tok.and_then(|tok| {
+                            let src_id = tok.get_src_id();
+
+                            let name = map.get_source(src_id);
+                            let content = map.get_source_contents(src_id);
+
+                            let (name, content) = name.zip(content)?;
+
+                            let path = VirtualFileSystem::new().root().join(name.into());
+                            let content =
+                                AssetContent::file(FileContent::new(File::from(content)).cell());
+
+                            Some(Vc::upcast(VirtualSource::new(path, content)))
+                        });
+                    }
+                }
+
+                token
+            }
+
+            SourceMap::Sectioned(map) => {
+                let len = map.sections.len();
+                let mut low = 0;
+                let mut high = len;
+                let pos = SourcePos { line, column };
+
+                // A "greatest lower bound" binary search. We're looking for the closest section
+                // offset <= to our line/col.
+                while low < high {
+                    let mid = (low + high) / 2;
+                    if pos < map.sections[mid].offset {
+                        high = mid;
+                    } else {
+                        low = mid + 1;
+                    }
+                }
+
+                // Our GLB search will return the section immediately to the right of the
+                // section we actually want to recurse into, because the binary search does not
+                // early exit on an exact match (it'll `low = mid + 1`).
+                if low > 0 && low <= len {
+                    let SourceMapSection { map, offset } = &map.sections[low - 1];
+                    // We're looking for the position `l` lines into region covered by this
+                    // sourcemap's section.
+                    let l = line - offset.line;
+                    // The source map starts offset by the section's column only on its first line.
+                    // On the 2nd+ line, the source map covers starting at column 0.
+                    let c = if line == offset.line {
+                        column - offset.column
+                    } else {
+                        column
+                    };
+
+                    return map
+                        .await?
+                        .lookup_token_and_source_internal(l, c, need_source_content)
+                        .await;
```

Comment:
You should call the turbo-tasks functions to allow caching on the nested source maps.

```suggestion
                    if need_source_content {
                        return map.lookup_token_and_source(l, c);
                    } else {
                        return (map.lookup_token_and_source(l, c), None);
                    }
```